lib/srv/db/postgres/schema.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v4"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	dbobjectv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1"
+	"github.com/gravitational/teleport/api/types/databaseobject"
+	"github.com/gravitational/teleport/lib/srv/db/common"
+	"github.com/gravitational/teleport/lib/srv/db/common/permissions"
+)
+
+// schemaInfo represents information about all schemas in a database.
+type schemaInfo map[string]schema
+
+// schema represents single schema.
+type schema struct {
+	tables []string
+}
+
+// schemaInfoQuery is a query to return the schema info.
+const schemaInfoQuery = "SELECT schemaname, tablename FROM pg_tables"
+
+func fetchDatabaseObjects(ctx context.Context, session *common.Session, conn *pgx.Conn) ([]*dbobjectv1.DatabaseObject, error) {
+	s, err := getSchemaInfo(ctx, conn)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var out []*dbobjectv1.DatabaseObject
+
+	for schemaName, schemaVal := range s {
+		for _, table := range schemaVal.tables {
+			name := strings.Join([]string{
+				session.Database.GetProtocol(),
+				session.Database.GetType(),
+				session.Database.GetName(),
+				permissions.ObjectKindTable,
+				session.DatabaseName,
+				schemaName,
+				table,
+			}, "/")
+
+			obj, err := databaseobject.NewDatabaseObject(name, &dbobjectv1.DatabaseObjectSpec{
```

Comment:
I've explicitly dropped that part of RFD to simplify the feature. I was considering adding a default import rule, but I think we can rather add it as an example in the docs.

Reply:
Specifically, I can add this rule if there are no other rules at auth startup?

```
kind: db_object_import_rule
metadata:
  name: import_all_objects
  namespace: default
spec:
  database_labels:
  - name: '*'
    values:
    - '*'
  mappings:
  - add_labels:
      kind: table
    match:
      table_names:
      - '*'
  - add_labels:
      kind: view
    match:
      view_names:
      - '*'
  - add_labels:
      kind: procedure
    match:
      procedure_names:
      - '*'
  priority: 0
version: v1
```

Reply:
Added: f42071b1ce7f715afdca919e49d48c4315d8d43d. PTAL?

Reply:
👍 ha, that's an interesting way to add kind labels!!

In my case though, I was hoping we auto-label things like `database_name`, `table_name`, etc. I initially had a single matcher in import rule to import tables from TWO databases, but I only wanted to grant tables from ONE of them to my role. To workaround the current limit, I had to split the matcher to two where each only matches one `database_name`, and explicitly add `database_name` labels. Lets not worry about this case now since it's not part of the RFD. We do auto-label things during discovery. So we could consider this in the future.

Reply:
Fair enough, I'll leave it as it is for now. I was considering adding an interpolation into the label value, so you could have a rule like this:

```yaml
kind: db_object_import_rule
metadata:
  name: tag_with_database_name
  namespace: default
spec:
  database_labels:
  - name: '*'
    values:
    - '*'
  mappings:
  - add_labels:
      kind: database_name
    match:
      table_names:
      - '{{object.database_name}}'
  priority: 100
version: v1
```